mat44xmat44.c

```diff
@@ -0,0 +1,115 @@
+#include <immintrin.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/time.h>
+
+#define N 4
+
+int
+main() {
+
+    // allocate memory for array
+    int are_equal = 1;
+    int count = 0;
+    float(*A)[N];
+    if (posix_memalign((void **) &A, 16, sizeof(float[N][N])) != 0) {
+        fprintf(stderr, "error allocating memory for matrix B\n");
+        exit(1);
+    }
+
+    float(*B)[N];
+    if (posix_memalign((void **) &B, 16, sizeof(float[N][N])) != 0) {
+        fprintf(stderr, "error allocating memory for matrix B\n");
+        exit(1);
+    }
+
+    float(*result)[N];
+    if (posix_memalign((void **) &result, 16, sizeof(float[N][N])) != 0) {
+        fprintf(stderr, "error allocating memory for matrix B\n");
+        exit(1);
+    }
+
+    float(*implResult)[N];
+    if (posix_memalign((void **) &implResult, 16, sizeof(float[N][N])) != 0) {
+        fprintf(stderr, "error allocating memory for matrix B\n");
+        exit(1);
+    }
+    // fill vector with numbers
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            A[i][j] = (float) count++;
+        }
+    }
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            B[i][j] = (float) count--;
+        }
+    }
+
+    struct timeval tv1, tv2, tv3, diff1, diff2;
+    gettimeofday(&tv1, NULL);
+
+    // normal scalar version
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            result[i][j] = 0.0f;
+            for (int k = 0; k < N; k++) {
+                result[i][j] += A[i][k] * B[k][j];
+            }
+        }
+    }
+    gettimeofday(&tv2, NULL);
+
+    //
+
+    for (int i = 0; i < N; i++) {
+        for (int j = 0; j < N; j++) {
+            implResult[i][j] = 0.0f;
+            for (int k = 0; k < N; k++) {
+                implResult[i][j] += A[i][k] * B[k][j];
+            }
+        }
+    }
+    gettimeofday(&tv3, NULL);
```

Comment:
what are you measuring here? this is the same loop. you could just add the SSE code here or if this is supposed to be a placeholder, just add a comment about that. Right now it seems to compare two identical loops but without an explanation.